apps/whale-api/src/module.api/poolpair.fees.service.e2e.ts

```diff
@@ -0,0 +1,399 @@
+import { DeFiDCache } from './cache/defid.cache'
+import { Testing } from '@defichain/jellyfish-testing'
+
+const container = new MasterNodeRegTestContainer()
+let app: NestFastifyApplication
+let controller: PoolPairController
+let testing: Testing
+
+beforeAll(async () => {
+  await container.start()
+  await container.waitForWalletCoinbaseMaturity()
+  await container.waitForWalletBalanceGTE(100)
+  await setup()
+
+  app = await createTestingApp(container)
+  controller = app.get(PoolPairController)
+  const defiCache = app.get(DeFiDCache)
+
+  const tokenResult = await container.call('listtokens')
+  // precache
+  for (const k in tokenResult) {
+    await defiCache.getTokenInfo(k)
+  }
+})
+
+afterAll(async () => {
+  await stopTestingApp(container, app)
+})
+
+async function setup (): Promise<void> {
+  testing = Testing.create(container)
+
+  const tokens = ['DUSD', 'CAT', 'DOG', 'KOALA', 'FISH']
+
+  for (const token of tokens) {
+    await createToken(container, token)
+    await createPoolPair(container, token, 'DFI')
+    await mintTokens(container, token)
+    await testing.generate(1)
+  }
+
+  await addPoolLiquidity(container, {
+    tokenA: 'DUSD',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 10,
+    shareAddress: await getNewAddress(container)
+  })
+
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'CAT',
+    amountA: 100,
+    tokenB: 'DFI',
+    amountB: 1000,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'DOG',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'KOALA',
+    amountA: 10,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  await addPoolLiquidity(container, {
+    tokenA: 'FISH',
+    amountA: 5,
+    tokenB: 'DFI',
+    amountB: 100,
+    shareAddress: await getNewAddress(container)
+  })
+  await testing.generate(1)
+
+  // dex fee set up
+  await container.call('setgov', [{
+    ATTRIBUTES: {
+      // CAT-DFI
+      'v0/poolpairs/4/token_a_fee_pct': '0.001', // CAT
+      'v0/poolpairs/4/token_a_fee_direction': 'in', // CAT
+      'v0/poolpairs/4/token_b_fee_pct': '0.002', // DFI
+      'v0/poolpairs/4/token_b_fee_direction': 'in', // DFI
+
+      'v0/poolpairs/6/token_a_fee_pct': '0.003', // DOG
+      'v0/poolpairs/6/token_a_fee_direction': 'out', // DOG
+      'v0/poolpairs/6/token_b_fee_pct': '0.004', // DFI
+      'v0/poolpairs/6/token_b_fee_direction': 'out', // DFI
+
+      'v0/poolpairs/8/token_a_fee_pct': '0.005', // KOALA
+      'v0/poolpairs/8/token_a_fee_direction': 'in', // KOALA
+
+      'v0/poolpairs/10/token_b_fee_pct': '0.006', // FISH
+      'v0/poolpairs/10/token_b_fee_direction': 'out' // FISH
```

Comment:
here needs to test the `both` as well
eg:
```
'v0/poolpairs/{id}/token_b_fee_pct': '0.007',
'v0/poolpairs/{id}/token_b_fee_direction': 'both'
```

Reply:
Good catch! Updated!